packages/admin/admin/src/form/FinalFormNumberInput.tsx

```diff
@@ -0,0 +1,80 @@
+import { InputBase, InputBaseProps } from "@mui/material";
+import * as React from "react";
+import { FieldRenderProps } from "react-final-form";
+import { useIntl } from "react-intl";
+
+import { ClearInputAdornment } from "../common/ClearInputAdornment";
+
+export type FinalFormNumberInputProps = InputBaseProps &
+    FieldRenderProps<number | string> & {
+        clearable?: boolean;
+        decimals?: number;
+    };
+
+export function FinalFormNumberInput({
+    meta,
+    input,
+    innerRef,
+    clearable,
+    endAdornment,
+    decimals = 0,
+    ...props
+}: FinalFormNumberInputProps): React.ReactElement {
+    const intl = useIntl();
+
+    const numberParts = intl.formatNumberToParts(1111.111);
+    const decimalSymbol = numberParts.find(({ type }) => type === "decimal")?.value;
+    const thousandSeparatorSymbol = numberParts.find(({ type }) => type === "group")?.value;
+
+    const [formattedNumberValue, setFormattedNumberValue] = React.useState("");
+
+    const getFormattedValue = React.useCallback(
+        (value, decimals: number | undefined) => {
+            const formattedValue =
+                value !== undefined ? intl.formatNumber(value, { minimumFractionDigits: decimals, maximumFractionDigits: decimals }) : "";
+            return formattedValue;
+        },
+        [intl],
+    );
+
+    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        const { value } = event.target;
+        setFormattedNumberValue(value);
+    };
+
+    const handleBlur = (event: React.FocusEvent<HTMLInputElement>) => {
+        const { value } = event.target;
+
+        const numericValue = parseFloat(value.split(`${thousandSeparatorSymbol}`).join("").split(`${decimalSymbol}`).join("."));
+
+        const inputValue = isNaN(numericValue) ? undefined : numericValue;
+        input.onChange(inputValue);
+        setFormattedNumberValue(getFormattedValue(inputValue, decimals));
+    };
```

Comment:
When I type in a decimal number, when `decimals === 0`, the decimal number is still written into the form-state even though the value shown is not a decimal number. 

https://github.com/vivid-planet/comet/assets/6264317/4f17aac2-a8f9-4f0c-a5b5-26ece8bde3fe

Reply:
This still seems to be the case.

Reply:
👍 
Now I got it - thank you for the video. I missunderstood your comment. 
I used toFixed now, to specify the decimals.